app/models/user.rb

```diff
@@ -1,2 +1,15 @@
 class User < ApplicationRecord
+  has_secure_password
+
+  has_many :folders
+  has_many :uploads, through: :folders
+
+  validates :username,        presence: true, uniqueness: true
+  validates :cellphone,       presence: true, uniqueness: true
+  validates :email,           presence: true, uniqueness: true
+  validates :password_digest, presence: true
+
+  def root_folder
```

Comment:
🎉 @AELSchauer @JonathanSR @alfosco let's remember to add flash messages to the log-in/account creation around these validations.